### ISSUE TITLE
Run Lumpy structural variant calling within BCBio with prior evidence BED/VCF file. 

### ISSUE THREAD

**Issue by NeillGibson** [bcbio/bcbio-nextgen#1729]
Hi, 

The Lumpy paper mentions that it's possible to feed "prior knowledge" about SV's to Lumpy to increase sensitivity. 

https://genomebiology.biomedcentral.com/articles/10.1186/gb-2014-15-6-r84

Is is possible to run Lumpy from within BCBio with a prior evidence BED file or VCF file?

On our test data Lumpy currently misses some known deletion SV's that we would like to have called and typed over a small cohort of a couple of hundred samples. 

Running Lumpy together with CNVKit would be another option, but I am afraid the computational cost of running CNVKit on a couple of hundred WGS samples is too high. Therefore it would be nice to have Lumpy working with the prior evidence signal, so that CNVKit only needs to be run on a few samples for discovery and Lumpy can be used for typing those SV's on larger cohorts. 

I looked in the bcbio SV calling documentation but I did not see a generic option for providing a prior evidence BED/VCF file. 
http://bcbio-nextgen.readthedocs.io/en/latest/contents/configuration.html#structural-variant-calling

Thank you.

**Comment by chapmanb** [comment]
Neill -- thanks for the suggestion. We don't have a way to provide a prior evidence file for Lumpy right now, but could include this. Right now we support CNVkit as evidence to Lumpy, as you mention, but no generic mechanism. Thanks again for starting this discussion.

**Comment by NeillGibson** [comment]
Hi Brad. Thank you for considering adding this feature. 

I still need to test running with Lumpy with CNVKit to see what the actual compute cost is and if it solves the missing deletion. 

In the end we would like to have 1 VCF with the called structural variants, even if multiple SV calling tools are used.  

Do I understand it correctly that only CNVKit is currently passed on to Lumpy?
Manta SV calls are for instance not passed on to Lumpy. 

On our data Lumpy is much more sensitivity than Manta for all types of known SVs, except that Manta is more sensitive for known insertions.  So it would be nice if all or just the insertion calls of Manta can be passed on to Lumpy. 

Another question that I have is if MetaSV is still part of bcbio? It is not mentioned in the SV calling documentation but can be found in the code. 
http://bcbio-nextgen.readthedocs.io/en/latest/contents/configuration.html#structural-variant-calling
https://github.com/chapmanb/bcbio-nextgen/blob/master/bcbio/structural/metasv.py

MetaSV can do something similar to Lumpy, aggregate SV and type SV calls from different SV callers? Has MetaSV been replaced by Lumpy? 

Thank you.

**Comment by chapmanb** [comment]
Neill;
Thanks for the questions, you're right we only pass CNVkit calls to Lumpy. I'd have a lot of interest in your evaluation of if this is worthwhile. It entails a run cost since CNVkit needs to run before Lumpy, rather than in parallel, and we've not yet evaluated how much it helps resolve missing calls. If it helps we could look at incorporating other callers, but it's a pretty big time investment to re-work and test so would like to make sure it seems worthwhile.

Practically we've had good luck calling with multiple callers and using a prioritization-based approach to look at these together, rather than trying to merge into a single VCF:

http://biorxiv.org/content/early/2016/11/04/084640
http://bcbio-nextgen.readthedocs.io/en/latest/contents/configuration.html#structural-variant-calling

`metasv` may still work in bcbio but we've not used it recently so removed it from the documentation. It provided some benefit but was slow and required work to maintain so we switched over to the prioritization based approach which in our hands did a better job of identifying useful variants.

On the SV side our current focus is on trying out linked read technologies to 10x to see if they help with resolution of events. We're hoping this, or long reads like PacBio or Oxford Nanopore, might help with resolving better truth sets to be able to further explore short-read only algorithms. Hope this helps.

**Comment by NeillGibson** [comment]
Hi Brad,  I'll let you know if including CNVKit fixes the missing deletion call and what I think about the run time. 

I am also not sure if passing the VCF's from other callers on to Lumpy is worth the effort. 

Thank you for the link and the paper, I will check them out. 

We are also interested in the use of synthetic and actual long read technology for more sensitive and specific SV discovery. 

That also fits with this feature request, it would be nice if the SV's discovered with long read tech can be typed on larger cohorts with short read data and callers.

**Comment by roryk** [comment]
Closing this out as its old and seems like the discussion has run its course. Thanks everyone!
